chain/src/main/java/org/ethereum/beacon/chain/pool/StatefulProcessor.java

```diff
@@ -0,0 +1,24 @@
+package org.ethereum.beacon.chain.pool;
+
+/**
+ * Stateful processor.
+ *
+ * <p>A processor that requires particular inner state to be initialized before it could be safely
+ * called by its clients.
+ *
+ * <p>{@link #isInitialized()} method indicates whether processor has already been initialized or
+ * not. It's a client responsibility to check {@link #isInitialized()} result before calling to the
+ * instance of this processor.
+ *
+ * <p>Implementor MAY throw an {@link AssertionError} if it's been called before inner state has
+ * been initialised.
+ */
+public interface StatefulProcessor {
```

Comment:
Not sure I understand the purpose of this interface. 
You either `assert` that a component is inited or just skip an entry if not inited. Couldn't this be the internal logic of this component?

Reply:
For instance, we have checkers which returns value of `boolean` type saying whether check passed or not. These checkers may have several inputs and state initialized from either one or all of those inputs. Since, check MUST returns `true/false` there is no room for merely discarding the value fed to a checker. We might use `Optional<Boolean>` for that but I am not sure that it would be a better choice.